app/controllers/home_controller.rb

```diff
@@ -14,20 +14,16 @@ def index
   def redirect_unauthenticated_to_permalinks!
     return if user_signed_in?
 
-    redirect_to(PermalinkRedirector.new(request.path).redirect_path || default_redirect_path)
-  end
+    redirect_path = PermalinkRedirector.new(request.path).redirect_path
 
-  def default_redirect_path
-    if request.path.start_with?('/web') || whitelist_mode?
-      new_user_session_path
-    elsif single_user_mode?
-      short_account_path(Account.local.without_suspended.where('id > 0').first)
-    else
-      about_path
-    end
+    redirect_to(redirect_path) if redirect_path.present?
```

Comment:
This uses `/web/explore` as a landing page for single-user instances and instances in limited federation mode too, which makes absolutely no sense to me.